spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/actions/RewriteDataFilesSparkAction.java

```diff
@@ -377,7 +377,8 @@ private Builder doExecuteWithPartialProgress(
     // stop commit service
     commitService.close();
 
-    int failedCommits = maxCommits - commitService.succeededCommits();
+    int totalCommits = groupsPerCommit == 1 ? ctx.totalGroupCount() : maxCommits;
```

Comment:
Probably my fault here but i'm not sure I understand this. Shouldn't we be doing totalGroupCount / groupsPerCommit ?

Reply:
Or we could do Min(totalGroupCount, maxCommits) ?